### ISSUE TITLE
Enable DNS 'accepted_payload_size' to 8192 by default

### ISSUE THREAD

**Issue by bedis** [haproxy/haproxy#346]
I saw multiple times people using the runtime resolver and complaining that the accepted size is too short for their workload or that it had tragic consequences (#341 

So I'd like to propose to set by default  the 'accepted_payload_size' parameter to enable EDNS and announce to the server that HAProxy can accept payloads bigger that 512 bytes (8192 in such case).

Before doing this, we have to evaluate the following points:
* check if all DNS servers support EDNS
* how to detect a server does not support EDNS
* what would be the mechanism to disable it on the fly for the next request

About the doc, mention of 'hold obsolete' would be useful to keep server configured with unseen response record for some period of time.

**Comment by git001** [comment]
Maybe the https://ednscomp.isc.org/ednscomp?zone=haproxy.org code can help to get some idea how a edns check could be done.

I see mostly in the containerized setup that the **accepted_payload_size** is increased.

**Comment by lukastribus** [comment]
I disagree, by making the default way larger we will trigger fragmentation issues, causing again issues and new traps for the users. We really *really* need TCP support instead.

The DNS operators are pushing resolvers to lower the default edns payload size to 1232 bytes, to avoid IP fragmentation and push for TCP support everywhere for the DNS Flag Day 2020:
https://dnsflagday.net/2020/

If you think it makes a big difference, push the default to 1232 bytes, but please do not make it larger than that.

We won't have reliable DNS resolution in Haproxy until TCP support is implemented. There is just no way around it.

**Comment by jfree-fs** [comment]
FWIW (4096@AWS):
https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/DNSBehavior.html
Maximum Response Size
If the resolver supports EDNS0 (as defined in RFC 2671), and advertises the EDNS0 option to Amazon Route 53, Route 53 permits responses up to 4096 bytes over UDP, without truncation.

**Comment by bedis** [comment]
Thank you all for those feedback!
I'll see how we can improve the situation but still remain as safe and reliable as possible. This may end up in sticking to 1232 bytes by default as suggested by @lukastribus
